libraries/botbuilder-ai-orchestrator/src/orchestratorRecognizer.ts

```diff
@@ -337,100 +345,136 @@ export class OrchestratorRecognizer extends AdaptiveRecognizer implements Orches
         return properties;
     }
 
-    private _initializeModel() {
-        if (!this._modelFolder) {
+    private _initializeModel(modelFolder?: string, snapshotFile?: string, resolverExternal?: LabelResolver): void {
+        if (this._resolver) {
+            return;
+        }
+        if (resolverExternal) {
+            this._resolver = resolverExternal;
+            this._isResolverExternal = true;
+            return;
+        }
+
+        if (!modelFolder) {
             throw new Error(`Missing "ModelFolder" information.`);
         }
 
-        if (!this._snapshotFile) {
+        if (!snapshotFile) {
             throw new Error(`Missing "ShapshotFile" information.`);
         }
 
-        if (!OrchestratorRecognizer.orchestrator && !this._resolver) {
-            // Create orchestrator core
-            const fullModelFolder = resolve(this._modelFolder);
-            if (!existsSync(fullModelFolder)) {
-                throw new Error(`Model folder does not exist at ${fullModelFolder}.`);
-            }
+        // Create orchestrator core
+        const fullModelFolder: string = resolve(modelFolder);
+
+        this._orchestrator = OrchestratorRecognizer.orchestratorMap.has(fullModelFolder)
+            ? OrchestratorRecognizer.orchestratorMap.get(fullModelFolder)
+            : ((): OrchestratorDictionaryEntry => {
```

Comment:
Why an IIFE?
```typescript
this._orchestrator = OrchestratorRecognizer.orchestratorMap.get();
if (!this._orchestrator) {
  /// contents of IIFE
}
```

Reply:
No idea, inherited from the original code. Not in scope for this bug-fixing PR.